src/base_circuitpython/terminal_handler.py

```diff
@@ -0,0 +1,108 @@
+from PIL import Image
+import threading
+import os
+import base64
+from io import BytesIO
+import time
+import collections
+import pathlib
+
+from common import utils
+import board
+import base_cp_constants as CONSTANTS
+import displayio
+import terminalio
+
+
+class Terminal:
+    def __init__(self):
+        self.__output_values = collections.deque()
+        self.__lock = threading.Lock()
+        self.__abs_path = pathlib.Path(__file__).parent.absolute()
+        self.__base_img = Image.open(
+            os.path.join(self.__abs_path, CONSTANTS.IMG_DIR_NAME, CONSTANTS.BLINKA_BMP)
+        )
+
+    def __create_newline(self, str_list):
+        self.__lock.acquire()
+        for string in str_list:
+            self.__output_values.appendleft(string)
+
+        over = len(self.__output_values) - CONSTANTS.CLUE_TERMINAL_LINE_NUM_MAX
+
+        # max CONSTANTS.CLUE_TERMINAL_LINE_NUM_MAX items in output_values
+        if over > 0:
+            for i in range(over):
+                self.__output_values.pop()
+
+        self.__lock.release()
+
+    def configure(self, no_verif=False):

```

Comment:
This method name is a bit confusing. Judging from the name `configure`, I wouldn't expect it to also draw things.

Reply:
ok, I'll look into changing it c: